src/Workspaces/CSharpTest/EmbeddedLanguages/RegularExpressions/CSharpRegexParserTests_TestGeneration.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#if false
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.RegularExpressions;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.RegularExpressions
+{
+    internal class LogicalStringComparer : IComparer<string>
+    {
+        [DllImport("shlwapi.dll", CharSet = CharSet.Unicode)]
+        private static extern int StrCmpLogicalW(string psz1, string psz2);
+
+        public static readonly IComparer<string> Instance = new LogicalStringComparer();
+
+        private LogicalStringComparer()
+        {
+        }
+
+        public int Compare(string x, string y)
+        {
+            return StrCmpLogicalW(x, y);
+        }
+    }
+
+    public class Fixture : IDisposable
+    {
+        public void Dispose()
+        {
+            var other = new Dictionary<string, string>();
+
+            var tree = SyntaxFactory.ParseSyntaxTree(
+                File.ReadAllText(@"C:\GitHub\roslyn-internal\Open\src\Workspaces\CSharpTest\RegularExpressions\CSharpRegexParserTests_BasicTests.cs"));
+
+            var methodNames = tree.GetRoot().DescendantNodes().OfType<MethodDeclarationSyntax>().Select(m => m.Identifier.ValueText);
+            var nameToIndex = new Dictionary<string, int>();
+
+            var index = 0;
+            foreach (var name in methodNames)
+            {
+                nameToIndex[name] = index;
+                index++;
+            }
+
+#if true
+            var tests =
+                CSharpRegexParserTests.nameToTest.Where(kvp => !kvp.Key.StartsWith("NegativeTest") && !kvp.Key.StartsWith("Reference"))
+                     .OrderBy(kvp => nameToIndex[kvp.Key])
+                     .Select(kvp => kvp.Value);
+#elif false
+            var tests =
+                CSharpRegexParserTests.nameToTest.Where(kvp => kvp.Key.StartsWith("NegativeTest"))
+                     .OrderBy(kvp => kvp.Key, LogicalStringComparer.Instance)
+                     .Select(kvp => kvp.Value);
+#else
+            var tests =
+                CSharpRegexParserTests.nameToTest.Where(kvp => kvp.Key.StartsWith("Reference"))
+                     .OrderBy(kvp => kvp.Key, LogicalStringComparer.Instance)
+                     .Select(kvp => kvp.Value);
+#endif
+
+
+            var val = string.Join("\r\n", tests);
+        }
+    }
+
+    [CollectionDefinition(nameof(MyCollection))]
+    public class MyCollection : ICollectionFixture<Fixture>
+    {
+    }
+
+    [Collection(nameof(MyCollection))]
+    public partial class CSharpRegexParserTests
```

Comment:
>CSharpRegexParserTests [](start = 25, length = 22)

From reading the doc on [Collection](https://gist.github.com/bradwilson/8423477), I expected to find some tests (annotated with `[Fact]`) in this class.
Is this meant to only be run manually?

Reply:
Yup.  This was all done manyally for generatin tests.  (hence _TestGeneration).

Reply:
Ok. It seems useful to keep this file, but please add some documentation.

---
In reply to: [212121143](https://github.com/dotnet/roslyn/pull/25520#discussion_r212121143) [](ancestors = 212121143)

Reply:
Given that this code already provides a way of regenerating the baselines. What do you think of better representing spans?

---
In reply to: [212122968](https://github.com/dotnet/roslyn/pull/25520#discussion_r212122968) [](ancestors = 212122968,212121143)